### DISCUSSION TITLE
How to get demo x,y,z value back to in-game coordinate on custom map.

### DISCUSSION THREAD

**Discussion by agpmilli** [saul/demofile#237] [Questions] [answered]
Hi,

I am working on my own custom map, it means that I cannot use the .json you linked above for the coordinate translation. When I am on my map and I use "getpos" it outputs
`-1363.000000 1094.050049 65.093811`  
as my position but at the same moment in the demo I obtain  
`3299500032 1149813120 1065615360`

What should I use to do the mapping from one to the other?

Even if I can get pos_x, pos_y and scale and use these formulas to find my map_x and map_y:
map_x = (worldSpace.x - pos_x) / scale
map_y = (worldSpace.y - pos_y) / scale
I still have some trouble understanding how you can use them to obtain Z, pitch, yaw and roll in function of the map scaling and position

I attach the demo such that you can have a look. (had to put it in a zip such that github accepts it)
[alolpat_21_0078.zip](https://github.com/saul/demofile/files/1901816/alolpat_21_0078.zip)

**Comment by saul** [comment]
The map_x and y variables are points on the overview image, not the world space coordinates within the level. I think the naming of the variables has mislead you there.

Can you show me the code you’re using to get the player’s position.

**Comment by agpmilli** [comment, marked as answer]
Basically what I do is I look at when entity enters the game and I had them to my `players` and then I do this to get the position:

``` 
demoFile.on('tickend', e => {
    if (players) {
	for (var i = 0; i < players.length; i++) {
	    let ent = demoFile.entities.getByUserId(players[i]);
	    logger.info(demoFile.currentTime, demoFile.currentTick, ent.name, ent.position);
        }
    }
}); 
```

And I obtain the results shown in the previous message.

**Comment by saul** [comment]
How about if you iterate through `demoFile.entities.players` instead of maintaining your own `players` array?

**Comment by agpmilli** [comment]
Just tried it, doesn't change the output. Should I directly obtain `-1363.000000 1094.050049 65.093811` as values?

**Comment by saul** [comment]
Yes, you should. It surprises me that the numbers you get from this library are integers - they should be floats. I’ll dig into this tonight.

**Comment by agpmilli** [comment]
Ok, it looks like it was a problem with my custom map and I now get the desired values.
